tests/gis_tests/relatedapp/tests.py

```diff
@@ -304,6 +306,176 @@ def test_collect(self):
+        self.assertIsNone(city.parcel_border_non_existant)
+        self.assertEqual(city.parcel_border_no_filter, (0.0, 0.0, 32.0, 32.0))
+
+    @skipUnlessDBFeature("supports_union_aggr")
+    def test_union_filter(self):
+        qs = City.objects.annotate(
+            parcel_border_union=Union(
+                "parcel__border2",
+                filter=~Q(parcel__name__icontains="ignore"),
+            ),
+            parcel_border_non_existant=Union(
+                "parcel__border2",
+                filter=Q(parcel__name__icontains="not present"),
+            ),
+            parcel_border_union_single=Union(
+                "parcel__border2",
+                filter=Q(parcel__name__contains="Alpha"),
+            ),
+        )
+        city = qs.get(name="Aurora")
+        self.assertEqual(
+            city.parcel_border_union.wkt,
+            "MULTIPOLYGON (((0 0, 5 5, 8 8, 0 0)), ((10 10, 15 15, 22 22, 10 10)))",
+        )
+        self.assertIsNone(city.parcel_border_non_existant)
+        self.assertEqual(
+            city.parcel_border_union_single.wkt, "POLYGON ((0 0, 5 5, 8 8, 0 0))"
+        )
+
+    @skipUnlessDBFeature("supports_union_aggr")
+    def test_union(self):
+        qs = City.objects.annotate(
+            parcel_border_union=Union(
+                "parcel__border2",
+            ),
+        )
+        city = qs.get(name="Aurora")
+        self.assertEqual(
+            city.parcel_border_union.wkt,
+            "MULTIPOLYGON (((0 0, 5 5, 8 8, 0 0)), ((10 10, 15 15, 22 22, 10 10)), "
+            "((24 23, 25 25, 32 32, 24 23)))",
+        )
+
+    @skipUnlessDBFeature("supports_union_aggr")
+    def test_union_filter_both(self):
+        qs = City.objects.annotate(
+            parcel_border_union=Union(
+                "parcel__border2",
+                filter=~Q(parcel__name__icontains="ignore"),
+            ),
+        )
+        city = qs.get(name="Aurora")
+        self.assertEqual(
+            city.parcel_border_union.wkt,
+            "MULTIPOLYGON (((0 0, 5 5, 8 8, 0 0)), ((10 10, 15 15, 22 22, 10 10)))",
+        )
+
+    @skipUnlessDBFeature("supports_union_aggr")
+    def test_union_filter_non_existant(self):
+        qs = City.objects.annotate(
+            parcel_border_non_existant=Union(
+                "parcel__border2",
+                filter=Q(parcel__name__icontains="not present"),
+            )
+        )
+        city = qs.get(name="Aurora")
+        self.assertIsNone(city.parcel_border_non_existant)
+
+    @skipUnlessDBFeature("supports_union_aggr")
+    def test_union_filter_single(self):
+        qs = City.objects.annotate(
+            parcel_border_union_single=Union(
+                "parcel__border2",
+                filter=Q(parcel__name__contains="Alpha"),
+            ),
+        )
+        city = qs.get(name="Aurora")
+        self.assertEqual(
+            city.parcel_border_union_single.wkt, "POLYGON ((0 0, 5 5, 8 8, 0 0))"
+        )
```

Comment:
These are already tested in `test_union_filter()`. Removed.

Reply:
yes I had only split the test to check which annotate was failing